### PR TITLE
Turret Fix up

### LINKED ISSUE #150: Turret Subsystems fixups
Should return in radians, but returns rotations
https://github.com/frc6377/crescendo_2024/blob/94524676095a5b952250268340ad2c5d587a3919/src/main/java/frc/robot/subsystems/TurretSubsystem.java#L173

The zeroTurret() method doesn't work in simulation, and would be hard to setup in simulation. Do only when Robot.isReal()
https://github.com/frc6377/crescendo_2024/blob/94524676095a5b952250268340ad2c5d587a3919/src/main/java/frc/robot/subsystems/TurretSubsystem.java#L153

This needs to be related to the gear ratio of the sensor tracking the turret position to be accurate. i.e. Need to do the reverse operation of calculateTurretPosition()
https://github.com/frc6377/crescendo_2024/blob/94524676095a5b952250268340ad2c5d587a3919/src/main/java/frc/robot/subsystems/TurretSubsystem.java#L412-L414

Use setInverted() to invert the motor output instead of negative sign here 
https://github.com/frc6377/crescendo_2024/blob/94524676095a5b952250268340ad2c5d587a3919/src/main/java/frc/robot/subsystems/TurretSubsystem.java#L305-L310

### PR DESCRIPTION
**Pull request by KianChilds**
## Justification
Fixes #150 

## Implementaion
Added a unit conversion
Added suggested reality check
Fixed turret rotation report
Fixed PID output

## Testing Done
Runs simulation

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/TurretSubsystem.java [hunk 1]
```diff
@@ -302,7 +306,7 @@ private static double fineTuneTurretRotation(
   private void setTurretPos(double setpoint) {
     turretGoalPositionEntry.log(setpoint);
     turretMotor.set(
-        -turretPIDController.calculate(
+        turretPIDController.calculate(
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/TurretSubsystem.java:309]: Do you need to initialize turretMotor with setInverted()? Or does this work reversed on the real turret vs the proto-turret?

Comment (KianChilds) [reply src/main/java/frc/robot/subsystems/TurretSubsystem.java:310]: addressed in the next commit

File: src/main/java/frc/robot/subsystems/TurretSubsystem.java [hunk 2]
```diff
@@ -409,9 +414,7 @@ public void simulationPeriodic() {
     turretSim.update(Robot.defaultPeriodSecs);
     turretAngleSim.setAngle(Math.toDegrees(turretSim.getAngleRads()));
     SmartDashboard.putNumber("Turret Angle", Math.toDegrees(turretSim.getAngleRads()));
-    simTurretPos.set(
-        Units.radiansToRotations(
-            turretSim.getAngleRads() / Constants.TurretConstants.TURRET_MOTOR_TURRET_RATIO));
+    simTurretPos.set(Units.radiansToRotations(turretSim.getAngleRads()));
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/TurretSubsystem.java:417]: This needs to be converted from the turret gear to the low gear rotations (i.e. multiplied by LOW_GEAR_CAN_CODER_RATIO)

**Review by azvanderpas — CHANGES_REQUESTED** [review]
Also seeing there's duplicate turret Commands getting setup in RobotContainer, probably from a incorrectly done merge.

File: src/main/java/frc/robot/subsystems/TurretSubsystem.java (hunk 2 above)
Comment (aetherstorm-dev) [reply src/main/java/frc/robot/subsystems/TurretSubsystem.java]: Resolved in 6e163f1

**Comment by aetherstorm-dev** [comment]
> Also seeing there's duplicate turret Commands getting setup in RobotContainer, probably from a incorrectly done merge.

Resolved in 6e163f1

**Review by caseyjbrotherton — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/TurretSubsystem.java [hunk 3]
```diff
@@ -585,7 +589,14 @@ public void simulationPeriodic() {
     turretSim.update(Robot.defaultPeriodSecs);
     turretAngleSim.setAngle(Math.toDegrees(turretSim.getAngleRads()));
     SmartDashboard.putNumber("Turret Angle", Math.toDegrees(turretSim.getAngleRads()));
-    simTurretPos.set(Units.radiansToRotations(turretSim.getAngleRads()));
+    simTurretPos.set(
+        Units.radiansToRotations(
+            turretSim.getAngleRads() / Constants.TurretConstants.TURRET_CONVERSION_FACTOR));
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/TurretSubsystem.java:594]:
```suggestion
            turretSim.getAngleRads() * Constants.TurretConstants.LOW_GEAR_CAN_CODER_RATIO));
```

Comment (aetherstorm-dev) [reply src/main/java/frc/robot/subsystems/TurretSubsystem.java]: Resolved in 6bc7649 (twice, I think)

**Review by azvanderpas — APPROVED** [review]

**Pull request merged**